mantidproject/mantid
# Fix issues in LoadMask

Fix two issues of `LoadMask`:

- Detector ID buffers need to be cleared, otherwise reuse will see previously loaded masked detectors.
- Memory leak of Poco::XML::Document.

See also issue description.

**To test:**

Code review.

Fixes #21450.

Not reported by a user, so no release notes, in particular since this is probably not observable with the GUI or the *functional* Python algorithm interface (need to create algorithm instance yourself to see the issue).

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
